test/functional/ui/screen.lua

```diff
@@ -385,6 +400,60 @@ function Screen:_current_screen()
   return table.concat(rv, '\n')
 end
 
+function Screen:snapshot_util(attrs, ignore)
+  -- util to generate screen test
+  pcall(function() self:wait(function() return "error" end, 250) end)
+  if ignore == nil then
+    ignore = self._default_attr_ignore
+  end
+  if attrs == nil then
+    attrs = {}
+    if self._default_attr_ids ~= nil then
+      for i, a in ipairs(self._default_attr_ids) do
+        attrs[i] = a
+      end
+    end
+
+    for i = 1, self._height do
+      local row = self._rows[i]
+      for j = 1, self._width do
+        local attr = row[j].attrs
+        if attr_index(attrs, attr) == nil and attr_index(ignore, attr) == nil then
+          table.insert(attrs, attr)
+        end
+      end
+    end
+  end
+
+  local rv = {}
+  for i = 1, self._height do
+    table.insert(rv, "  "..self:_row_repr(self._rows[i],attrs, ignore).."|")
+  end
+  local attrstrs = {}
+  local alldefault = true
+  for i, a in ipairs(attrs) do
+    if self._default_attr_ids == nil or self._default_attr_ids[i] ~= a then
+      alldefault = false
+    end
+    local items = {}
+    for f, v in pairs(a) do
+      table.insert(items, f.." = "..tostring(v))
+    end
+    local dict = "{"..table.concat(items, ", ").."}"
+    table.insert(attrstrs, "["..tostring(i).."] = "..dict)
+    ::continue::
```

Comment:
Did you forget to remove this?

In any case, plain lua does not support `::continue::` so it is best to remove it(and keep the ability to run functional tests without luajit as it is better for debugging).